sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1786,7 +1786,25 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper with Logging {
    * Both un-targeted (global) and targeted aliases are supported.
    */
   override def visitStar(ctx: StarContext): Expression = withOrigin(ctx) {
-    UnresolvedStar(Option(ctx.qualifiedName()).map(_.identifier.asScala.map(_.getText).toSeq))
+    if (ctx.exceptClause != null) {
+      visitStarExcept(ctx)
+    }
+    else {
+      UnresolvedStar(Option(ctx.qualifiedName()).map(_.identifier.asScala.map(_.getText).toSeq))
```

Comment:
nit: we can put `Option(ctx.qualifiedName()).map(_.identifier.asScala.map(_.getText).toSeq)` into a variable to save duplicated code with `visitStarExcept`.